### PR TITLE
Document more list repos behavior

### PR DESCRIPTION
**Pull request by Wauplin**
Related to https://github.com/huggingface/huggingface_hub/issues/1814.

This PR updates the documentation of `ModelInfo`, `DatasetInfo`, `SpaceInfo` and `RepoSiblings` to let the users know they should not expect all attributes to exist. @ademait could you have a look and let me know what you think? I kept the wording a bit vague on purpose as I don't want to be too specific on things that we might change server-side in the future (all those objects are built in a way that is future-compatible, on purpose). (**EDIT:** I'm not saying that we will remove attributes in the future, but we might potentially add news ones and/or improve default behaviors).

### CODE REVIEW

File: src/huggingface_hub/hf_api.py [hunk 1]
```diff
@@ -1341,7 +1375,7 @@ def list_models(
 
         >>> # List only models from the AllenNLP library
         >>> api.list_models(filter="allennlp")
-
+        ```
```
Comment (julien-c) [inline src/huggingface_hub/hf_api.py:1378]: ~~intended?~~ yes i guess so

Comment (Wauplin) [reply src/huggingface_hub/hf_api.py:1378]: Yes I just realized [docs](https://huggingface.co/docs/huggingface_hub/v0.19.1/en/package_reference/hf_api#huggingface_hub.HfApi.list_models) looks awful at the moment :smile:

**Comment by ademait** [comment]
Hi @Wauplin , thanks for addressing the issue! I guess know it is more clear 🙌

**Comment by Wauplin** [comment]
Good thanks for the feedback :)
A final review from the team and we'll be good to merge.

File: src/huggingface_hub/hf_api.py [hunk 2]
```diff
@@ -378,6 +378,16 @@ class RepoSibling:
     """
     Contains basic information about a repo file inside a repo on the Hub.
 
+    <Tip>
+
+    All attributes of this class are optional except `rfilename`. This is because only the file names are returned when
+    listing repositories on the Hub (with [`list_models`], [`list_datasets`] or [`list_spaces`]). If you need more
+    information like file size, blob id or lfs details, you must request them specifically for one repo at a time
```
Comment (stevhliu) [inline src/huggingface_hub/hf_api.py:385]:
```suggestion
    information like file size, blob id or lfs details, you must request them specifically from one repo at a time
```

File: src/huggingface_hub/hf_api.py [hunk 3]
```diff
@@ -488,6 +498,14 @@ class ModelInfo:
     """
     Contains information about a model on the Hub.
 
+    <Tip>
+
+    Most attributes of this class are optional. This is because the data returned by the Hub depends on the query made.
+    In general, the more specific the query, the more information is returned. On the contrary, when listing models
+    using [`list_models`] only a subset of the attributes are returned.
```
[the remark this answers is not in the document]
Comment (Wauplin) [reply src/huggingface_hub/hf_api.py:505]: `ModelInfo` is the class returned by `list_models` (_get info from many_) and `model_info` (_get info from one_) but is not really meant to be used/instantiated outside of this scope. That's why I took the liberty to tie the documentation of ModelInfo to list_models.

**Review by LysandreJik — APPROVED** [review]
Thank you @Wauplin

File: src/huggingface_hub/hf_api.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (stevhliu) [reply src/huggingface_hub/hf_api.py:505]: Ah ok, thanks for the clarification. In that case, feel free to ignore this comment! :)

**Comment by Wauplin** [comment]
Thanks for the reviews!

**Pull request merged**